app/com/linkedin/drelephant/schedulers/OozieScheduler.java

```diff
@@ -0,0 +1,184 @@
+    schedulerName = schedulerConfData.getSchedulerName();
+
+    if (properties != null && properties.getProperty(OOZIE_ACTION_ID) != null) {
+      oozieClient = makeOozieClient(schedulerConfData);
+      jobDefUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_JOB_DEF_URL_TEMPLATE);
+      jobExecUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_JOB_EXEC_URL_TEMPLATE);
+      workflowDefUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_WORKFLOW_DEF_URL_TEMPLATE);
+      workflowExecUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_WORKFLOW_EXEC_URL_TEMPLATE);
+
+      loadInfo(appId, properties);
+    } else {
+      // Use default value of data type
+    }
+  }
+
+  private void loadInfo(String appId, Properties properties) {
+    // 0004167-160629080632562-oozie-oozi-W@some-action
+    jobExecId = properties.getProperty(OOZIE_ACTION_ID);
+    if (jobExecId.contains("@")) {
+      String[] pair = jobExecId.split("@", 2);
+      workflowExecId = pair[0];
+      jobName = pair[1];
+
+      try {
+        logger.info("Fetching Oozie workflow info for " + workflowExecId);
+        WorkflowJob workflow = oozieClient.getJobInfo(workflowExecId);
+        logger.info("Oozie workflow for " + workflowExecId + ": " + workflow);
+        workflowName = workflow.getAppName();
+      } catch (OozieClientException e) {
+        throw new RuntimeException("Failed fetching Oozie workflow " + workflowExecId + " info", e);
+      }
+    }
+
+    workflowDepth = 0; // TODO: Add sub-workflow support
+  }
+
+  private OozieClient makeOozieClient(SchedulerConfigurationData schedulerConfData) {
+    String oozieApiUrl = schedulerConfData.getParamMap().get(OOZIE_API_URL);
+
+    if (oozieApiUrl == null) {
+      throw new RuntimeException("Missing " + OOZIE_API_URL + " param for Oozie Scheduler");
+    }
+
+    return new OozieClient(oozieApiUrl);
+  }
+
+  @Override
+  public String getSchedulerName() {
+    return schedulerName;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return schedulerName == null || jobName == null || jobExecId == null || workflowName == null || workflowExecId == null;
+  }
+
+  @Override
+  public String getJobDefId() {
+    return Utils.formatStringOrNull("%s", jobName);
+  }
+
+  @Override
+  public String getJobExecId() {
+    return Utils.formatStringOrNull("%s", jobExecId);
+  }
+
+  @Override
+  public String getFlowDefId() {
+    return Utils.formatStringOrNull("%s", workflowName);
+  }
+
+  @Override
+  public String getFlowExecId() {
+    return Utils.formatStringOrNull("%s", workflowExecId);
+  }
+
+  @Override
+  public String getJobDefUrl() {
+    if (jobDefUrlTemplate != null) {
+      logger.warn("Missing " + OOZIE_JOB_DEF_URL_TEMPLATE + " param for Oozie Scheduler");
+      return Utils.formatStringOrNull(jobDefUrlTemplate, jobName);
+    } else {
+      return jobName;
+    }
+  }
+
+  @Override
+  public String getJobExecUrl() {
+    if (jobExecUrlTemplate != null) {
```

Comment:
Oozie doesn't have default UI (one can use Hue, CLI or built custom UI around an API), hence I can't force people to provide these URLs.

However, maybe they should be null in case there are no URLs configured instead of plain IDs as I've done now?